pkg/deviceshifu/deviceshifusocket/deviceshifusocket_test.go

```diff
@@ -86,64 +83,45 @@ func TestDeviceHealthHandler(t *testing.T) {
 	}
 }
 
-func TestCreateHTTPCommandlineRequestString(t *testing.T) {
-	req, err := http.NewRequest("GET", "http://localhost:8081/start?time=10:00:00&flags_no_parameter=-a,-c,--no-dependency&target=machine2", nil)
-	klog.Infof("%v", req.URL.Query())
-	createdRequestString := createHTTPCommandlineRequestString(req, "/usr/local/bin/python /usr/src/driver/python-car-driver.py", "start")
+func TestPraseCommandWithEncode(t *testing.T) {
+	input := "1230000abc"
+	var outputHex = []byte{18, 48, 0, 10, 188}
+
+	output, err := praseCommandWithEncode(input, EncodeHexStr)
 	if err != nil {
-		t.Errorf("Cannot create HTTP commandline request: %v", err.Error())
+		t.Errorf("%+v", err)
```

Comment:
Let's print out the error context in addition to the error itself, same for other errors in test